pkg/kubelet/kubelet.go

```diff
@@ -1287,7 +1286,8 @@ func (kl *Kubelet) getPullSecretsForPod(pod *api.Pod) ([]api.Secret, error) {
 	for _, secretRef := range pod.Spec.ImagePullSecrets {
 		secret, err := kl.kubeClient.Secrets(pod.Namespace).Get(secretRef.Name)
 		if err != nil {
-			return nil, err
+			glog.Warningf("Unable to retrieve pull secret %s/%s for %s/%s due to %v.  The image pull may not succeed.", pod.Namespace, secretRef.Name, pod.Namespace, pod.Name, err)
```

Comment:
> maybe just tolerate not found errors?

The real distinction being?  If it fails, syncPod will try again later, right?  Other types of errors should be transient or the whole kubelet is going to have trouble, right?